drivers/entropy/entropy_stm32.c

```diff
@@ -103,17 +103,46 @@ static void configure_rng(void)
 {
 	RNG_TypeDef *rng = entropy_stm32_rng_data.rng;
 
+#ifdef STM32_CONDRST_SUPPORT
+#if DT_INST_NODE_HAS_PROP(0, nist_config)
+	/*
+	 * Configure the RNG_CR in compliance with the NIST SP800.
+	 * The nist-config is direclty copied from the DTS.
+	 * The RNG clock must be 48MHz else the clock DIV is not adpated.
+	 * The RNG_CR_CONDRST is set to 1 at the same time the RNG_CR is written
+	 */
+#if defined(RNG_CR_ARDIS)
+	 /* For STM32U5 series, the ARDIS bit7 is considered in the nist-config */
+	MODIFY_REG(rng->CR,
+		RNG_CR_NISTC | RNG_CR_CLKDIV | RNG_CR_RNG_CONFIG1 |
+		RNG_CR_RNG_CONFIG2 | RNG_CR_RNG_CONFIG3 | RNG_CR_ARDIS,
+		(uint32_t)(RNG_CR_CONDRST | DT_INST_PROP(0, nist_config)));
+#else /* RNG_CR_ARDIS */
+	MODIFY_REG(rng->CR,
+		RNG_CR_NISTC | RNG_CR_CLKDIV | RNG_CR_RNG_CONFIG1 |
+		RNG_CR_RNG_CONFIG2 | RNG_CR_RNG_CONFIG3,
+		(uint32_t)(RNG_CR_CONDRST | DT_INST_PROP(0, nist_config)));
+#endif /* RNG_CR_ARDIS */
+#else /* nist_config */
+	LL_RNG_EnableCondReset(rng);
```

Comment:
`LL_RNG_EnableCondReset(rng)` is executed only if `#if DT_INST_NODE_HAS_PROP(0, nist_config)`.
Is it then correct to always do `LL_RNG_DisableCondReset(rng);` (_lines [141, 144]_) even if `nist_config` is not present?